pairinteraction/CMakeLists.txt

```diff
@@ -113,19 +113,111 @@ set_target_properties(wignerSymbols PROPERTIES POSITION_INDEPENDENT_CODE ON)
 target_link_libraries(pireal PRIVATE wignerSymbols)
 target_link_libraries(picomplex PRIVATE wignerSymbols)
 
-# Include Eigen
+# Eigen
 find_package(Eigen3 REQUIRED NO_MODULE)
 target_link_libraries(pireal PUBLIC Eigen3::Eigen)
 target_link_libraries(picomplex PUBLIC Eigen3::Eigen)
 
+# Find MKL
+set(MKL_ARCH "intel64" CACHE STRING "MKL architecture")
+set(MKL_LINK "dynamic" CACHE STRING "MKL linkage")
+set(MKL_THREADING "sequential" CACHE STRING "MKL threading backend")
+set(MKL_INTERFACE "lp64" CACHE STRING "MKL interface")
+find_package(MKL CONFIG)
+find_package(PkgConfig)
+if (NOT MKL_FOUND AND PKG_CONFIG_FOUND)
+  if(MKL_LINK STREQUAL "sdl")
+    pkg_check_modules(MKL mkl-sdl IMPORTED_TARGET)
+  elseif(MKL_THREADING STREQUAL "sequential")
+    pkg_check_modules(MKL mkl-${MKL_LINK}-${MKL_INTERFACE}-seq IMPORTED_TARGET)
+  elseif(MKL_THREADING STREQUAL "intel_thread")
+    pkg_check_modules(MKL mkl-${MKL_LINK}-${MKL_INTERFACE}-iomp IMPORTED_TARGET)
+  elseif(MKL_THREADING STREQUAL "gnu_thread")
+    pkg_check_modules(MKL mkl-${MKL_LINK}-${MKL_INTERFACE}-gomp IMPORTED_TARGET)
+  elseif(MKL_THREADING STREQUAL "tbb_thread")
+    pkg_check_modules(MKL mkl-${MKL_LINK}-${MKL_INTERFACE}-tbb IMPORTED_TARGET)
+  endif()
+endif()
+
+# Find a different lapack vendor
+find_package(LAPACK)
+
+# Determine which lapack vendor to use
+if(WITH_LAPACK AND MKL_FOUND AND (NOT BLA_VENDOR OR BLA_VENDOR MATCHES "All" OR (BLA_VENDOR MATCHES "^Intel")))
```

Comment:
This condition is completely unreadable and surely does not work correctly, because `BLA_VENDOR MATCHES "All"` might also select MKL if there is no other LAPACK implementation. To be quite honest I think the code is now worse that before.

Reply:
The code did not work at all before. It did not provide any working option to select whether one prefers to use MKL or OpenBLAS. MKL was always used as soon as MKL was installed.

The condition ensures that MKL is used only if a) MKL is found and lapack should be used *and* b) no vendor is specified that does not start with "Intel". `BLA_VENDOR MATCHES "All"` is necessary because cmake sets `BLA_VENDOR` to `All` if it is not specified by the user. At least, this was the behavior of my cmake version.